dexter/src/main/java/com/karumi/dexter/AndroidPermissionService.java

```diff
@@ -18,22 +18,47 @@
 
 import android.app.Activity;
 import android.content.Context;
+import android.content.pm.PackageManager;
 import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
 import android.support.v4.app.ActivityCompat;
 import android.support.v4.content.ContextCompat;
 
+import static com.karumi.dexter.AndroidVersionPermissionMap.isHandledPermission;
+
 /**
  * Wrapper class for all the static calls to the Android permission system
  */
 class AndroidPermissionService {
 
+    private final int sdkVersion;
+
+    AndroidPermissionService(int sdkVersion) {
+        this.sdkVersion = sdkVersion;
+    }
+
   /**
+   * Support library ContextCompat.checkSelfPermission handling unknown permissions
    * @see ContextCompat#checkSelfPermission
    */
   int checkSelfPermission(@NonNull Context context, @NonNull String permission) {
-    return ContextCompat.checkSelfPermission(context, permission);
+      if (isHandledPermission(sdkVersion, permission)) {
+          return supportCheckSelfPermission(context, permission);
+    } else {
+      return PackageManager.PERMISSION_GRANTED;
+    }
   }
 
+    /**
+     * Support library ContextCompat.checkSelfPermission
+     *
+     * @see ContextCompat#checkSelfPermission
+     */
+    @VisibleForTesting
```

Comment:
We don't like this annotation at all but after review another possible approaches to test this class using [some of the official android tools](http://tools.android.com/tech-docs/unit-testing-support#TOC-Method-...-not-mocked.-) I think the usage of Robolectric is going to best approach. 

We should remove this annotation, change this method visibility to private and change the tests we've written to use Robolectric with different SDK versions.